crypto/bls.go

```diff
@@ -338,7 +391,11 @@ func (sk *prKeyBLSBLS12381) Size() int {
 func (sk *prKeyBLSBLS12381) computePublicKey() {
 	var newPk pubKeyBLSBLS12381
 	// compute public key pk = g2^sk
-	generatorScalarMultG2(&(newPk.point), &(sk.scalar))
+	generatorScalarMultG2(&newPk.point, &sk.scalar)
+
+	// cache the identity comparison
+	newPk.isIdentity = C.bn_is_zero((*C.bn_st)(&sk.scalar)) != valid
```

Comment:
It would be nice to wrap this in a utility function, since we use it several times:
* `bn_is_zero`'s definition [doesn't define its return value in terms of the `VALID` constant](https://github.com/relic-toolkit/relic/blob/15cb6d1dbe03a2780aae27fe6b31682d9c2d0942/include/relic_bn.h#L444-L450)
* `bn_is_zero`'s "true" return value interpreted in terms of the `VALID` constant maps to `INVALID`, which was the opposite of what I'd expect

```suggestion
	newPk.isIdentity = bnIsZero(sk.scalar)
```

Reply:
> bn_is_zero's definition doesn't define its return value in terms of the VALID constant

Good point. `valid` constant is mapped to Relic's `RLC_OK` while `bn_is_zero` is simply using `0`.

Reply:
https://github.com/onflow/flow-go/pull/3421/commits/4f3197b9e3843ed4b693744fff1bd4b90174cdbc